switchbot/adv_parsers/keypad.py

```diff
@@ -0,0 +1,21 @@
+"""Keypad parser."""
+
+from __future__ import annotations
+
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def process_wokeypad(
+    data: bytes | None, mfr_data: bytes | None,
+) -> dict[str, bool | int | None]:
+    """Process woKeypad services data."""
+    if data is None or mfr_data is None:
+        return {"battery": None, "attemptState": None}
+
+    _LOGGER.debug("mfr_data: %s", mfr_data.hex())
+    if data:
+        _LOGGER.debug("data: %s", data.hex())
+
+    return {"battery": data[2], "attemptState": mfr_data[6]}
```

Comment:
Are you sure battery uses all the bits?
Most of the other devices need  `& 0b01111111`

Reply:
Nothing from the advertisement data besides the attempt state has changed as I've been using it so I don't know if the eigth bit is used for anything but it only needs 7 bits to make 100 anyway, I'll fix this